msu/systems/mod_settings/elements/color_picker_setting.nut

```diff
@@ -31,6 +31,20 @@
 		return asRGBA;
 	}
 
+	function getValueAsHexString()
+	{
+		local asArray = split(this.Value, ",");
+		local red = format("%x", asArray[0].tointeger());
+		local green = format("%x", asArray[1].tointeger());
+		local blue = format("%x", asArray[2].tointeger());
```

Comment:
Have you tested if this works? It will throw an exception if `tointeger()` returns an integer instead of a string:
```
sq>print(format("%s", 2))

AN ERROR HAS OCCURED [string expected for the specified format]

CALLSTACK
*FUNCTION [main()] interactive console line [1]

LOCALS
[vargv] ARRAY
[this] TABLE
```

```
sq>print(format("%s", "2"))
2
sq>
```

```
sq>print(format("%s", "2".tointeger()))

AN ERROR HAS OCCURED [string expected for the specified format]

CALLSTACK
*FUNCTION [main()] interactive console line [1]

LOCALS
[vargv] ARRAY
[this] TABLE
```

Reply:
%x, not %s. %x takes int and gives hex value.

Reply:
Ah, I see. Then looks good.